### ISSUE TITLE
Add support for async processor

### ISSUE THREAD

**Issue by Plawn** [asciidoctor/asciidoctor.js#1694]
I want to use async function in a processor to fetch an external resource but I can't seem to use an async function.

Having an async processor would make extending the base processor way easier

Thanks !

**Comment by ggrossetie** [comment]
I agree but Asciidoctor.js is transpiled from Ruby and basically it's not possible (very hard) to support `async` functions. You can take a look at https://github.com/ggrossetie/unxhr in order to make synchronous HTTP requests in Node.

**Comment by ggrossetie** [comment]
In the next version of Asciidoctor.js, processing will be asynchronous.

**Comment by Plawn** [comment]
dude :) nice
